Building with GCC 15 on ARM64 sometimes fail due to GCS

A number of build failures are being reported which relate to using GCC 15 on ARM64.  The compiler recently gained more capabilities w.r.t. [GCS](https://docs.kernel.org/arch/arm64/gcs.html), and that's interacting badly with `ld`.

For example, when trying to use `cgo`  on ARM64 with GCC 15, we currently see the following problem:

```
$ cat > 1.go << _EOF_
        package main
        /*
        #include <stdlib.h>
        #include <stdio.h>
        void hello() {
            printf("%s\\n", "Hello from cgo!");
            fflush(stdout);
        }
        */
        import "C"
        func main() {
            C.hello()
        }
_EOF_
$ go run 1.go
# command-line-arguments
/usr/lib/go/pkg/tool/linux_arm64/link: running gcc failed: exit status 1
/usr/local/bin/gcc -s -o $WORK/b001/exe/1 -rdynamic /tmp/go-link-624876020/go.o /tmp/go-link-624876020/000000.o /tmp/go-link-624876020/000001.o /tmp/go-link-624876020/000002.o /tmp/go-link-624876020/000003.o /tmp/go-link-624876020/000004.o /tmp/go-link-624876020/000005.o /tmp/go-link-624876020/000006.o /tmp/go-link-624876020/000007.o /tmp/go-link-624876020/000008.o /tmp/go-link-624876020/000009.o /tmp/go-link-624876020/000010.o /tmp/go-link-624876020/000011.o /tmp/go-link-624876020/000012.o /tmp/go-link-624876020/000013.o /tmp/go-link-624876020/000014.o /tmp/go-link-624876020/000015.o -O2 -g -O2 -g -lpthread
/tmp/go-link-624876020/go.o: warning: GCS is required by -z gcs, but this input object file lacks the necessary property note.
/tmp/go-link-624876020/000015.o: warning: GCS is required by -z gcs, but this input object file lacks the necessary property note.
/usr/lib/gcc/aarch64-unknown-linux-gnu/15/../../../../lib64/libgcc_s.so.1: warning: GCS is required by -z gcs, but this shared library lacks the necessary property note. The dynamic loader might not enable GCS or refuse to load the program unless all the shared library dependencies have the GCS marking.
/usr/lib/gcc/aarch64-unknown-linux-gnu/15/../../../../lib64/libgcc_s.so.1: warning: GCS is required by -z gcs, but this shared library lacks the necessary property note. The dynamic loader might not enable GCS or refuse to load the program unless all the shared library dependencies have the GCS marking.
/usr/lib/gcc/aarch64-unknown-linux-gnu/15/../../../../aarch64-unknown-linux-gnu/bin/ld: read-only segment has dynamic relocations
collect2: error: ld returned 1 exit status
```

--- a/openssf-compiler-options.yaml
+++ b/openssf-compiler-options.yaml
@@ -1,7 +1,7 @@
 package:
   name: openssf-compiler-options
   version: 20240627
-  epoch: 22
+  epoch: 23
   description: "Compiler Options Hardening Guide for C and C++"
   url: https://best.openssf.org/Compiler-Hardening-Guides/Compiler-Options-Hardening-Guide-for-C-and-C++.html
   copyright:

--- a/openssf-compiler-options/usr/lib/gcc/aarch64-unknown-linux-gnu/15/openssf.spec
+++ b/openssf-compiler-options/usr/lib/gcc/aarch64-unknown-linux-gnu/15/openssf.spec
@@ -2,7 +2,7 @@
 + %{!O:%{!O1:%{!O2:%{!O3:%{!O0:%{!Os:%{!0fast:%{!0g:%{!0z:-O2}}}}}}}}} -fhardened -Wno-error=hardened -Wno-hardened -mbranch-protection=standard %{!fdelete-null-pointer-checks:-fno-delete-null-pointer-checks} -fno-strict-overflow -fno-strict-aliasing %{!fomit-frame-pointer:-fno-omit-frame-pointer} -mno-omit-leaf-frame-pointer
 
 *link:
-+ --as-needed -O1 --sort-common -z noexecstack -z relro -z now
++ --as-needed -O1 --sort-common -z noexecstack -z relro -z now -z gcs=never
 
 %include_noerr </usr/lib/oldglibc/gcc.spec>
 %include_noerr </home/build/.melange.gcc.spec>

